internal/users/manager.go

```diff
@@ -52,7 +96,11 @@ func NewManager(config Config, cacheDir string) (m *Manager, err error) {
 	}
 
 	m = &Manager{
-		config: config,
+		config:                config,
+		uidsToGenerateInTests: opts.uidsToGenerateInTests,
+		gidsToGenerateInTests: opts.gidsToGenerateInTests,
```

Comment:
Can we instead get these differently? Like using a function that is defined only in `_test.go` files?

Reply:
I don't think so. We can't access functions defined in `_test.go` from the non-test `users` package.

Reply:
I don't want to use the `go:linkname` hack again

Reply:
I also don't see the problem with having the `WithUIDsToGenerateInTests` function here, since it name makes it clear that this is only for tests.

Reply:
I was thinking more on something like what we do with [export_test.go files](https://github.com/ubuntu/authd/blob/main/pam/internal/gdm/export_test.go)